src/libexpr/common-eval-args.cc

```diff
@@ -40,7 +41,16 @@ MixEvalArgs::MixEvalArgs()
       .description("override a flake registry value")
       .arity(2)
       .handler([&](std::vector<std::string> ss) {
-          registryOverrides.push_back(std::make_pair(ss[0], ss[1]));
+          registryOverrides.insert_or_assign(FlakeRef(ss[0]), FlakeRef(ss[1]));
+      });
+
+    mkFlag()
+      .longName("override-lockfile")
+      .labels({"original-ref", "resolved-ref"})
+      .description("override a lockfile value")
+      .arity(2)
+      .handler([&](std::vector<std::string> ss) {
+          lockFileOverrides.insert_or_assign(FlakeRef(ss[0]), FlakeRef(ss[1]));
```

Comment:
This probably should be added to `EvalCommand` rather than `MixEvalArgs`. The latter is also used by legacy commands (`nix-env` etc) which don't implement flake-related stuff.